### ISSUE TITLE
Accessibility: Dynamic content updates (filtering) not announced to screen readers

### ISSUE THREAD

**Issue by philippsauer** [intershop/intershop-pwa#1839]
**Actual Behavior**

When a category filter is applied on a category page, the displayed product list updates visually. However, screen reader users are not immediately informed of this change. Their focus remains on the filter controls, and they must manually tab through all remaining filters before discovering the updated product list. This leads to disorientation and the impression that no change occurred.

**Expected Behavior**

After applying a category filter, screen reader users should be immediately made aware of the content change and the updated results. This can be achieved by 

1 - automatically shifting the focus to the beginning of the new results (e.g., the results heading:  set tabindex=”-1” and focus via .focus().

2 - announcing a summary of the change using an ARIA live region, or ideally, a combination of 1 and 2.

For Example:
`<div aria-live=”polite” aria-atomic=”true”>`
`666 Ergebnisse für Schreibtischzubehör geladen.`
`</div>`

**Steps to Reproduce the Bug**

1. Activate NVDA
2. Go to https://intershoppwa-b2b.azurewebsites.net/computers/monitors/flat-panel-displays-ctgComputers.3001.222
3. Navigat to "Flat Panel Mounts" using your keyboard
4. The screen reader does not immediately announce that the product list has updated. The focus remains on the filter panel. The user only discovers the new results (in this case 0) by manually navigating the entire page content.

**Additional Context, like Screenshots, Log File Snippets etc.**

![Image](https://github.com/user-attachments/assets/d7275331-469e-48b7-a062-2625effafcf1)

[AB#107498](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/107498)
